### DISCUSSION TITLE
How to implement my QuickAccess Service?

### DISCUSSION THREAD

**Discussion by erickRochaIP** [CodinGame/monaco-vscode-api#476] [Q&A] [answered]
Hello!

I would like to know what is the correct way to override services so that I can filter which commands will be available in the Command Palette.
My goal is to prevent the user from having access to "dangerous commands", for example, installing extensions remotely.
Is it possible to be achieved using this api?

**Comment by CGNonofr** [comment, marked as answer]
I'm not sure just hiding the command from the command palette would be enough to ensure that the user can install remote extensions.

Maybe you can put the extension directory as readonly on the server side?

If you still want to just hide some command, there is no built-in way, but you can probably hack it with something like that:
```typescript
import { CommandsQuickAccessProvider } from '@codingame/monaco-vscode-quickaccess-service-override/vscode/vs/workbench/contrib/quickaccess/browser/commandsQuickAccess'

const original = CommandsQuickAccessProvider.prototype['getCommandPicks']
CommandsQuickAccessProvider.prototype['getCommandPicks'] = async function (this: CommandsQuickAccessProvider, token: CancellationToken) {
  let result = await original.call(this, token)
  return result.filter(commandPick => <whatever you want>)
}
```

**Reply by erickRochaIP** [reply]
I'm using codingame as a dependency of my project, and the provided import fails to resolve the dependency.
Using this:
```js
import { CommandsQuickAccessProvider } from '@codingame/monaco-vscode-quickaccess-service-override/vscode/src/vs/workbench/contrib/quickaccess/browser/commandsQuickAccess'
```
also fails to resolve the dependency, but matches the correct path to the `CommandsQuickAccessProvider` file.
Do you know how to resolve these issues with exporting dependencies?

**Reply by CGNonofr** [reply]
I think you did not copy/paste my snippet and made a typo :)

**Reply by erickRochaIP** [reply]
Just pointing out that it was necessary to add a `// @ts-ignore` before the import. I don't know why, but the IDE keeps complaining about the import even though it works perfectly when running.

**Reply by CGNonofr** [reply]
That's expected, the types are not available for that module
